### PR TITLE
Create FUTURE-NATIVE.md

### PR DESCRIPTION
**Pull request by jaredly**
@knowbody was asking about the future of the native side of things, figured I'd write it up here.
cc @wokalski @jordwalke @andreypopp @rauanmayemir

### CODE REVIEW

**Comment by knowbody** [comment]
Thank you for writing this down @jaredly!

cc @grabbou

File: FUTURE-NATIVE.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+# Potential direction #1
+status: not started
+
+Use existing react-native infrastructure, and make a reason-react implementation that talks the same protocol as react-native currently uses.
+
+Benefits: we can re-use all the flex-box, native view bindings, etc. that react-native already has.
+Downsides: that might come with some cruft as well.
+
+# Potential direction #2
+status: started (repo [here](https://github.com/pasmomusic/reason-react/pull/8))
```
Comment (wokalski) [inline FUTURE-NATIVE.md]: this has actually moved to https://github.com/briskml/brisk

File: FUTURE-NATIVE.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+# Potential direction #1
+status: not started
+
+Use existing react-native infrastructure, and make a reason-react implementation that talks the same protocol as react-native currently uses.
+
+Benefits: we can re-use all the flex-box, native view bindings, etc. that react-native already has.
+Downsides: that might come with some cruft as well.
+
+# Potential direction #2
```
Comment (wokalski) [inline FUTURE-NATIVE.md]: Actually I think that the step before is common for the direction number one and two. 

Brisk is separate from any platform specific things. The ambition for the project is to make it useful both when you want to use RN's `MessageQueue`, Terminal, or maybe the Mac. We're striving for the right abstraction level to be totally platform agnostic.

File: FUTURE-NATIVE.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
+Lots of people have asked about reason-react compiled to native (Reason Native React Native or RNRN 😎), here's a rough outline of dependencies that have to fall into place.
+
+# Prerequisites
+status: in progress
+
+The main thing here is getting a package manager + build toolchain that make cross-compilation a breeze. @jordwalke and some others are working on this right now, and it's called [esy](https://github.com/esy/esy).
+
+Time remaining before it's usable: 2-8 months? :P [TODO put in a better estimate about timeline]
```
Comment (rauanmayemir) [inline FUTURE-NATIVE.md]: Here's a PR you can follow for cross-target compilation progress: https://github.com/esy/esy/pull/169.
It seamlessly compiles any basic dune projects but doesn't work for older ocamlbuild setups without busting `OCAML_TOPLEVEL_PATH` (due to topfind constraints).

**Comment by prometheansacrifice** [comment]
Are [brisk](https://github.com/briskml/brisk) and [rereact-native](https://github.com/lpalmes/rereact-native) getting merged into reason-react?

cc @rauanmayemir @lpalmes

**Comment by jaredly** [comment]
I'll take "no activity" as "this is fine" :P

**Comment by jordwalke** [comment]
I set up [a separate repo](https://github.com/jordwalke/native-reason-react-roadmap) just for the sake of discussing, brainstorming and pointing to links of any existing efforts. No code should go there, but let's use it as a central place to organize and make sure separate projects are aware of existing work that they can reuse/contribute to. I'm excited about this idea, and apologies for taking so long to reply. We can merge this .md file into the main README.md in that repo.

**Comment by wokalski** [comment]
@jordwalke I’m primarily looking for inspiration. I mean ideas how to solve things like animations, state management, etc.

Since we are starting fresh and are cut off from the legacy of react Id like to improve in the areas where react is deficient.

**Pull request merged**